pkgs/by-name/pr/proton-ge-bin/package.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, stdenvNoCC
+, fetchzip
+, writeScript
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "proton-ge-bin";
+  version = "GE-Proton9-1";
+
+  src = fetchzip {
+    url = "https://github.com/GloriousEggroll/proton-ge-custom/releases/download/${finalAttrs.version}/${finalAttrs.version}.tar.gz";
+    hash = "sha256-odpzRlzW7MJGRcorRNo784Rh97ssViO70/1azHRggf0=";
+  };
+
+  outputs = [ "out" "steamcompattool" ];
+
+  buildCommand = ''
+    runHook preBuild
+
+    # Provide an error message when adding it to an env
+    echo "${finalAttrs.pname} should not be installed into environments. Please use programs.steam.extraCompatPackages." > $out
+
+    ln -s $src $steamcompattool
```

Comment:
just for me to understand what we are doing here.
Basically we are using fetchzip, that automatically extracts to $src, on  building the derivation we link $src to $steamcompattool.

A small question about the log, shouldnt that also be printed when building the derivation (overall) derivation in hydra later? I mean likely no one might care about that (just to understand it).

Reply:
> Basically we are using fetchzip, that automatically extracts to $src, on building the derivation we link $src to $steamcompattool.

Yes, in this case, that's the easiest method. Alternatively, we could copy everything over but I don't see the point in that.

Reply:
> A small question about the log, shouldnt that also be printed when building the derivation (overall) derivation in hydra later? I mean likely no one might care about that (just to understand it).

Which log are you talking about?

Reply:
> and as second question (related to [#295785 (comment)](https://github.com/NixOS/nixpkgs/pull/295785#discussion_r1524227440)):
> 
> > We could symlink that to $out/share/steam/compatibilitytools.d/$name/ too now that I think about it.
> 
> I assume that you dropped that idea, correct? (fine for me, just that i did not overlook something)

We could but I don't really see the point for NixOS purposes tbh.

We'd add `/run/current-system/sw/share/compatibilitytools.d/` to the extra compat paths by default and users would be able to add compat tools to the system-wide enironment.

That'd mostly be there fore users who cling to FHS-isms such as installing software that only one component actually needs into a global path. Making the global env fail to build when doing that simply makes it impossible to be in such a situation; forcing users to utilise the appropriate option.

The only unknown here is non-NixOS purposes where I could see that being useful. I'd say we cross that bridge when we get there and an actual user of that concept needs it.

Reply:
> Yes, in this case, that's the easiest method. Alternatively, we could copy everything over but I don't see the point in that.

One minor cosmetic issue this causes is that it's not obvious which version of proton is being ran when looking at `ps` and the like as it'd just say `/nix/store/0bcbzzxskaw13d378fv3630cv5k09kxf-source/proton waitforexitandrun ...`.

Reply:
> Which log are you talking about?

I assume that this should be shown in the build logs, isn't it? I mean buildCommand is run when building the derivation eG on hydra or?

Reply:
> I assume that this should be shown in the build logs, isn't it? I mean buildCommand is run when building the derivation eG on hydra or?

What is "this"?

Reply:
Ah nvm, the line was to long and it did not fit on my screen... so i did not see the ` > $out ` on the echo.
I thought that it would be a log when building, but with the pipe it make more sense to me.
Sorry for the noise.